test/auth.js

```diff
@@ -628,6 +628,30 @@ describe('authentication', function () {
             });
         });
 
+        it('matches scope while using request context (single to single)', function (done) {
+
+            var server = new Hapi.Server();
+            server.connection();
+            server.auth.scheme('custom', internals.implementation);
+            server.auth.strategy('default', 'custom', true, { users: { steve: { scope: 'one-test' } } });
+            server.route({
+                method: 'GET',
+                path: '/{id}',
+                config: {
+                    handler: function (request, reply) { return reply(request.auth.credentials.user); },
+                    auth: {
+                        scope: 'one-{params.id}'
```

Comment:
Need to test partial `{}` (e.g. one side missing), along with multiple `{}` in one scope if that feature is kept.